Add freerdp backend

LDprg opened the pull request:
This moved from #30 to here for better way of working.

oskardotglobal commented on winapps/src/freerdp.rs:
```diff
@@ -19,19 +19,24 @@ pub mod freerdp_back {
             xfreerdp.args([
-                //"/app:".to_owned() + app,
+                "/app:".to_owned() + app,
                 "/d:".to_owned() + &config.rdp.domain,
```
You should probably use [`format!`](https://doc.rust-lang.org/std/macro.format.html) for these.
Also, you can pass `&str` to these instead of `String`, see https://github.com/oskardotglobal/winapps/blob/rewrite/winapps/src/quickemu.rs#L34-L42

LDprg replied:
Thx, forgot about this one.

LDprg commented:
So I think generally the the base is finished and ready for merge. Just waiting for some reviews.

Maybe we want to change to command names (later this would we much more complicated).

oskardotglobal reviewed (approved):
Looks great.

oskardotglobal commented:
> Maybe we want to change to command names
We could just keep what you've called `app` as `manual`, but imo `run` would be better

AkechiShiro commented:
@oskardotglobal did you try and build this branch ? Or should I test it on my side ?

oskardotglobal commented:
Nope, I'll test this later

AkechiShiro commented:
I tested the build works, but there are a few issues, I noticed @LDprg

- `winapps-cli check` ![image](https://github.com/winapps-org/winapps/assets/14914796/dfb4e9c6-f62e-4036-a8c3-9bf6f00a25b0)
- I tried a few commands, but they seemed to fail if `XDG_CONFIG_HOME` wasn't defined, we should automatically fallback on `$HOME/.config`

oskardotglobal commented:
@AkechiShiro the latter is included in #26

LDprg commented:
@AkechiShiro The first one is that winapps doesnt disable the freerdp output (which containst alot of unnecessary warings). I will investigate into this.

LDprg commented:
@AkechiShiro did you test it with the generated quickemu from #26? I am not sure if we need a port forwarding from the guest to host to make rdp communication working per default.

AkechiShiro commented:
@LDprg no I removed all the previous files to test cleanly but I understand that there is no quickemu get and VMs files

oskardotglobal commented:
@LDprg I'll test it

LDprg commented:
So finally I fixed the log outputs of freerdp causing unessecary warnings to be shown. Later we could redirect its output to some log files, however this will be in a sperate PR (just want this get through as fast as possible). 

Furthermore I fixed some minor issues after the merge of #26 and renamed the app command to run.
@oskardotglobal can you review this so I can merge.

oskardotglobal commented on winapps/src/freerdp.rs:
```diff
@@ -6,20 +6,58 @@ pub mod freerdp_back {
+                Some(exe) => {
+                    xfreerdp.args([
+                        &format!("/app:{}", exe),
+                        "-wallpaper",
```
Any reason why you add the `-wallpaper` flag here and not if the app isn't given?

oskardotglobal reviewed (changes requested):
Looks good otherwise

LDprg commented:
@oskardotglobal I removed it. Can you approve so I can merge?

oskardotglobal reviewed (approved):
Looks good.
